README.md

```diff
@@ -1 +1,49 @@
-# opentelemetry-collector-operator
\ No newline at end of file
+# opentelemetry-collector-operator
+
+## SingletonSnapManager: Locking and Registration
```

Comment:
This is a great piece of documentation!
Since this is a charm repo, should we move this information into the `src/singleton_snap.py` module level docstring?